superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigForm/FiltersConfigForm.tsx

```diff
@@ -345,36 +347,45 @@ const FiltersConfigForm = (
       defaultValueQueriesData: null,
       isDataDirty: false,
     });
+    setError('');
```

Comment:
Might be cleaner to make a wrapper function `setFilterQueryResult` that calls both `setNativeFilterFieldValues` and `setError` since they are always called together.

As well as setting the value to null when there is an error maybe?